puppeteer-scraper/README.md

```diff
@@ -1,222 +1,492 @@
+-   [Additional Resources](#additional-resources)
+-   [Upgrading](#upgrading)
 
 <!-- tocstop -->
 
-## How it works
-Puppeteer Scraper is the most powerful scraper tool in our arsenal (aside from developing your own actors).
-It uses the Puppeteer library to programmatically control a headless Chrome browser and it can make it do
-almost anything. If using the Web Scraper does not cut it, Puppeteer Scraper is what you need.
+## Usage
+
+To get started with Puppeteer Scraper, you only need a few things. First, with `Start URLs`, tell the scraper which web pages it should load. Then, tell it how to handle each request and extract data from each page.
+
+The scraper starts by loading pages specified in the [**Start URLs**](#start-urls) input setting. You can make the scraper follow page links on the fly by setting a **[Link selector](#link-selector)** and/or **[Pseudo-URLs](#pseudo-urls)** to tell the scraper which links it should add to the crawler's request queue. This is useful for the recursive crawling of entire websites (e.g. finding all products available in an online store).
+
+To tell the scraper how to handle requests and extra data, you need to provide a **[Page function](#page-function)**, and optionally arrays of **[Pre-navigation hooks](#pre-navigation-hooks)** and **[Post-navigation hooks](#post-navigation-hooks)**. This is JavaScript code that is executed in the Node.js environment. Since the scraper uses the full-featured Chromium browser, client-side logic to be executed within the context of the web-page can be done using the **[`page`](#page)** object within the Page function's context.
+
+In summary, PuppeteerScraper works as follows:
+
+1. Adds each URL from [Start URLs](#start-urls) to the request queue.
+2. For each request:
+    - Evaluates all hooks in [Pre-navigation hooks](#pre-navigation-hooks)
+    - Executes the [Page function](#page-function) on the loaded page
+    - Optionally, finds all links from the page using [Link selector](#link-selector). If a link matches any of the [Pseudo URLs](#pseudo-urls) and has not yet been requested, it is added to the queue.
+    - Evaluates [Post-navigation hooks](#post-navigation-hooks)
+3. If there are more items in the queue, repeats step 2. Otherwise, finishes the crawl.
+
+Puppeteer Scraper has a number of other configuration settings to improve performance, set cookies for login to websites, mask the web browser, etc.. See [Advanced configuration](#advanced-configuration) below for the complete list of settings.
+
+## Limitations
+
+The actor employs a fully-featured Chromium web browser, which is resource-intensive and might be overkill for websites that do not render the content dynamically using client-side JavaScript. To achieve better performance for scraping such sites, you might prefer to use [**Cheerio Scraper**](https://apify.com/apify/cheerio-scraper), which downloads and processes raw HTML pages without the overheads of a full-on web browser.
+
```

Comment:
"full-on" is a common phrase in American English. Puppeteer uses a full-on straight up no-cap web browser instead of something that just sends regular HTTP requests like Axios or Cheerio ;)
I'll change this to something else